worker/src/main/scala/com/lucidchart/piezo/ConnectionProvider.scala

```diff
@@ -1,25 +1,162 @@
+      ipAddress, // Since we are connecting to writer nodes in AWS, we should connect using the IP address instead of CNAME
       props.getProperty("org.quartz.dataSource." + dataSource + ".user"),
       props.getProperty("org.quartz.dataSource." + dataSource + ".password"),
       props.getProperty("org.quartz.dataSource." + dataSource + ".maxConnections").toInt,
       props.getProperty("org.quartz.dataSource." + dataSource + ".validationQuery")
-    ))
-  } else {
-    logger.info("No job store found in config")
-    None
+    )
+  }
+
+  /**
+   * Its necessary to close HikariCP connection Pools when changes are made to AWS Aurora infrastructure.
+   *   - when we do a fail-over of the Writer instance from us-east-1 to us-east-2
+   *   - When we do a fail-over of the Writer instance within region by promoting another reader
+   *
+   * For Writer instances, this is achieved by keeping track of IP-address of the writer instance. Change in IP-address
+   * indicates both same region fail-over and cross-region
+   *
+   * NOTE: AWS Blue/Green Deployments automatically drop connections to the DB instances:
+   * https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/blue-green-deployments-switching.html#blue-green-deployments-switching-actions
+   * When that happens, the hikari pool
+   *   - Automatically closes the connections when they were closed remotely by AWS
+   *   - Automatically attempts to open up new connections to maintain the minimum # of idle connections. There is a
+   *     timeout for how long it waits for a connection to be established. If those connections timeout/get rejected by
+   *     AWS because the switchover isn't complete, the pool keeps attempting to open up new connections until the
+   *     minimum # is reached
+   *
+   * This function returns True, iff
+   *   - IP addresses have changed for the CNAME record used for DNS lookup
+   *
+   * @param pool
+   *   the connection pool currently being used
+   * @param dnsIP
+   *   the IP returned when performing a DNS lookup
+   * @return
+   */
+  private def mustKillConnectionPool(pool: Pool, dnsIP: String): Boolean = {
+    pool.ip != dnsip
+  }
+
+  /**
+   * Resolve the hostname to an ip address. Uses the built-in JVM cache but if that throws an exception, uses our cache
+   * with the last JVM value
+   *
+   * @param hostname - Comes from the config (which is set by lucidbag), which is a CNAME to a writer endpoint. Failovers for writer nodes
+   *                 are caused by a change in IP addresses, so a DNS lookup of the IP address will help us identify when a failover occurred
+   */
+  def getIP: String = {
+    Try {
+      // Get the ip address of the hostname.  The result is cached in the JVM
+      java.net.InetAddress.getByName(dataSourceHostname).getHostAddress
+    } match {
+      case Failure(e) => {
+        // Get the last cached version
+        Option(dnsCache.get(dataSourceHostname)).asInstanceOf[String] match {
```

Comment:
I actually think retries would probably be better than using a backup cache, if dns lookup fails.

Reply:
I'm impartial to either approach. I'll update it to a retry system